add noise and reverberance version for BinauralWSJ0Mix

huangzj421 opened the pull request:
Hi there, 
I have created a noise and reverberance version of BinauralWSJ0Mix datasets and trained with convtasnet-parallel structure. Here are the recipes and not conflicted with the clean version of datasets.
Also, I have trained convtasnet-parallel.yaml again and got a better results which I could share you with the Google Driver.
Thanks.

huangzj421 commented:
Hi @ycemsubakan , do you have time for checking this PR please?

ycemsubakan commented:
Hi @huangzj421 , sorry for the delay. I will try to do it soon.

ycemsubakan commented:
@huangzj421 , Could you add in the README the instructions on how to run the model with these noisy conditions? As before, the script should automatically download and create the corresponding dataset.

ycemsubakan commented:
For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.

![image](https://user-images.githubusercontent.com/17621727/182046029-21346761-b65a-4587-afe2-a48b88bb2b6e.png)

ycemsubakan commented:
And, I am observing the following error for `convtasnet-parallel-reverb.yaml`:

![image](https://user-images.githubusercontent.com/17621727/182046139-e34a6e49-82ed-4684-915f-49b4fe371e0d.png)

huangzj421 commented:
> For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> 
> ![image](https://user-images.githubusercontent.com/17621727/182046029-21346761-b65a-4587-afe2-a48b88bb2b6e.png)

I have corrected the data generation scripts and now these should work fine.

huangzj421 commented:
> @huangzj421 , Could you add in the README the instructions on how to run the model with these noisy conditions? As before, the script should automatically download and create the corresponding dataset. I would like to try myself if everything works correctly.

The usage is the same as the previous PR which you should only set wsj_root and output_root and it will download and generate data into output_root.

huangzj421 commented:
> And, I am observing the following error for `convtasnet-parallel-reverb.yaml`:
> 
> ![image](https://user-images.githubusercontent.com/17621727/182046139-e34a6e49-82ed-4684-915f-49b4fe371e0d.png)
> 
> Please note that I tried `convtasnet-parallel.yaml`, and that seems to work fine. Correction: Actually, I am also observing that the is stuck at 80, and it doesn't go down for `convtasnet-parallel.yaml'. In the previous PR, I tried and it the error was going down as expected.. I am not sure what's happening here..

I am sorry that I don't understand the "Correction" part you said. Also the test check failed but I don't know why.

ycemsubakan commented:
> 

Actually, if the dataset is already created, we should not require `wsj_root` again, right?

ycemsubakan commented:
> 

What I mean with the 'correction' part is that, when I run the script, (with `convtasnet_parallel.yaml`, the error does not go down. It is stuck at 80.

huangzj421 commented:
> > 
> 
> Actually, if the dataset is already created, we should not require `wsj_root` again, right?

Yes, if the dataset is done and dynamic mixing don't need it in the hparams["base_folder_dm"].

huangzj421 commented:
> 

OK, let me check it.

ycemsubakan commented:
> 
Yes, but we would also not need `wsj_root` if the dataset is already created.

ycemsubakan commented:
> > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > ![image](https://user-images.githubusercontent.com/17621727/182046029-21346761-b65a-4587-afe2-a48b88bb2b6e.png)
> 
> I have corrected the data generation scripts and now these should work fine.

Ok, can you push your code after this correction in the data generation script?

huangzj421 commented:
> > > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > > ![image](https://user-images.githubusercontent.com/17621727/182046029-21346761-b65a-4587-afe2-a48b88bb2b6e.png)
> > 
> > 
> > I have corrected the data generation scripts and now these should work fine.
> 
> Ok, can you push your code after this correction in the data generation script?

Oh, it doesn't need to push cause I just corrected in my [repo](https://github.com/huangzj421/Binaural-WSJ0Mix).

ycemsubakan commented:
> > > > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > > > ![image](https://user-images.githubusercontent.com/17621727/182046029-21346761-b65a-4587-afe2-a48b88bb2b6e.png)
> > > 
> > > 
> > > I have corrected the data generation scripts and now these should work fine.
> > 
> > 
> > Ok, can you push your code after this correction in the data generation script?
> 
> Oh, it doesn't need to push cause I just corrected in my [repo](https://github.com/huangzj421/Binaural-WSJ0Mix).

I am now getting this error: 

![image](https://user-images.githubusercontent.com/17621727/182068518-8c2f035e-0741-46fe-a47c-89c8c42ae046.png)

huangzj421 commented:
> > > > > For `convtasnet-parallel-noisy.yaml` I am getting this error. You can see the command I used also here.
> > > > > ![image](https://user-images.githubusercontent.com/17621727/182046029-21346761-b65a-4587-afe2-a48b88bb2b6e.png)
> > > > 
> > > > 
> > > > I have corrected the data generation scripts and now these should work fine.
> > > 
> > > 
> > > Ok, can you push your code after this correction in the data generation script?
> > 
> > 
> > Oh, it doesn't need to push cause I just corrected in my [repo](https://github.com/huangzj421/Binaural-WSJ0Mix).
> 
> I am now getting this error:
> 
> ![image](https://user-images.githubusercontent.com/17621727/182068518-8c2f035e-0741-46fe-a47c-89c8c42ae046.png)
It seems that the download [link](https://data.deepai.org/DemandDataset.zip) is invalid. I am trying to search another one.

huangzj421 commented:
> > 
> 
> What I mean with the 'correction' part is that, when I run the script, (with `convtasnet_parallel.yaml`, the error does not go down. It is stuck at 80.

What is the unit of 80? 80%?

huangzj421 commented:
Hi @ycemsubakan , I have fixes some errors in train.py and the data generation scripts in my repo. After you remove the previous BinauralWSJ0Mix folder and run train.py again, it will download automatically and train smoothly.
But I still don't know the 80 stuck you said and pytest check failure. Thanks.

ycemsubakan commented:
The noisy version seems to work after your last push. (the data generation) 

But the loss is still kind of stuck:
<img width="727" alt="Capture d’écran, le 2022-08-01 à 10 25 49" src="https://user-images.githubusercontent.com/17621727/182169958-f9c23ee7-460e-4e18-9442-5e2af6404b60.png">

huangzj421 commented:
> The noisy version seems to work after your last push. (the data generation)
> 
> But the loss is still kind of stuck: <img alt="Capture d’écran, le 2022-08-01 à 10 25 49" width="727" src="https://user-images.githubusercontent.com/17621727/182169958-f9c23ee7-460e-4e18-9442-5e2af6404b60.png">

It was weired cause I just trained and the loss goes down well:
![image](https://user-images.githubusercontent.com/73161104/182175918-902e9bf7-f87a-42cd-bd64-a8b3aa6a1a00.png)
My command was:
CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl'
Your loss was too big, did you use the dynamic mixing?

ycemsubakan commented:
> > The noisy version seems to work after your last push. (the data generation)
> > But the loss is still kind of stuck: <img alt="Capture d’écran, le 2022-08-01 à 10 25 49" width="727" src="https://user-images.githubusercontent.com/17621727/182169958-f9c23ee7-460e-4e18-9442-5e2af6404b60.png">
> 
> It was weired cause I just trained and the loss went down well: ![image](https://user-images.githubusercontent.com/73161104/182175918-902e9bf7-f87a-42cd-bd64-a8b3aa6a1a00.png) My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?

No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?

huangzj421 commented:
> > > The noisy version seems to work after your last push. (the data generation)
> > > But the loss is still kind of stuck: <img alt="Capture d’écran, le 2022-08-01 à 10 25 49" width="727" src="https://user-images.githubusercontent.com/17621727/182169958-f9c23ee7-460e-4e18-9442-5e2af6404b60.png">
> > 
> > 
> > It was weired cause I just trained and the loss went down well: ![image](https://user-images.githubusercontent.com/73161104/182175918-902e9bf7-f87a-42cd-bd64-a8b3aa6a1a00.png) My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?
> 
> No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?

It was not so big as well:
![image](https://user-images.githubusercontent.com/73161104/182177614-fbd1d454-47dc-45df-8b1a-5a4b28bd15e9.png)

ycemsubakan commented:
> 



> > > > The noisy version seems to work after your last push. (the data generation)
> > > > But the loss is still kind of stuck: <img alt="Capture d’écran, le 2022-08-01 à 10 25 49" width="727" src="https://user-images.githubusercontent.com/17621727/182169958-f9c23ee7-460e-4e18-9442-5e2af6404b60.png">
> > > 
> > > 
> > > It was weired cause I just trained and the loss went down well: ![image](https://user-images.githubusercontent.com/73161104/182175918-902e9bf7-f87a-42cd-bd64-a8b3aa6a1a00.png) My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?
> > 
> > 
> > No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?
> 
> It was not so big as well: ![image](https://user-images.githubusercontent.com/73161104/182177614-fbd1d454-47dc-45df-8b1a-5a4b28bd15e9.png)

Hm.. weird.. I am not sure why this is happening when I try. This doesn't happen with other recipes. I'll try to understand what's happening.

ycemsubakan commented:
> > > 
> > 
> > 
> > > > > > The noisy version seems to work after your last push. (the data generation)
> > > > > > But the loss is still kind of stuck: <img alt="Capture d’écran, le 2022-08-01 à 10 25 49" width="727" src="https://user-images.githubusercontent.com/17621727/182169958-f9c23ee7-460e-4e18-9442-5e2af6404b60.png">
> > > > > 
> > > > > 
> > > > > It was weired cause I just trained and the loss went down well: ![image](https://user-images.githubusercontent.com/73161104/182175918-902e9bf7-f87a-42cd-bd64-a8b3aa6a1a00.png) My command was: CUDA_VISIBLE_DEVICES=3,5,6,7 python -m torch.distributed.launch --nproc_per_node=4 train.py hparams/convtasnet-parallel.yaml --data_folder /mnt/test/ --wsj_root /mnt/wsj/ --distributed_launch --distributed_backend='nccl' Your loss was too big, did you use the dynamic mixing?
> > > > 
> > > > 
> > > > No, I am not using dynamic mixing. I am training on a single gpu. Can you try training on a single gpu?
> > > 
> > > 
> > > It was not so big as well: ![image](https://user-images.githubusercontent.com/73161104/182177614-fbd1d454-47dc-45df-8b1a-5a4b28bd15e9.png)
> > 
> > 
> > Hm.. weird.. I am not sure why this is happening when I try. This doesn't happen with other recipes. I'll try to understand what's happening.
> 
> Maybe you could remove all the dataset and train it again?

Yes, I already tried that. I generated the dataset from scratch.

huangzj421 commented:
Hi @ycemsubakan, do you have any updates?

ycemsubakan commented:
@huangzj421 No, I will try to get to it this week.

huangzj421 commented:
> @huangzj421 No, I will try to get to it this week.

Okay, thanks!

ycemsubakan commented:
With `convtasnet-parallel-noise.yaml`, I observed the following error. Do you see it also? 

![image](https://user-imagesgithubusercontent.com/17621727/183339455-0ecc0fc7-5817-4167-bad8-f68e391845b4.png)

huangzj421 commented:
> With `convtasnet-parallel-noise.yaml`, I observed the following error. Do you see it also?
> 
> ![image](https://user-images.githubusercontent.com/17621727/183339582-81f91087-0b94-4e12-a9a9-4388684c444d.png)

No, I guess you should update the automatic downloaded data generation scripts from https://github.com/huangzj421/Binaural-WSJ0Mix or you can remove the old version in your data_folder.

ycemsubakan commented:
> > With `convtasnet-parallel-noise.yaml`, I observed the following error. Do you see it also?
> > ![image](https://user-images.githubusercontent.com/17621727/183339582-81f91087-0b94-4e12-a9a9-4388684c444d.png)
> 
> No, I guess you should update the automatic downloaded data generation scripts from https://github.com/huangzj421/Binaural-WSJ0Mix or you can remove the old version in your data_folder.

I see.. It was working earlier. But, this could be an issue for people who use this recipe in the future. Can you automatically check if the folders are correct, and then if not, you can either automatically fix, or at least you need to give an error message so that people know that they should delete the folder, and create a new one.

huangzj421 commented on recipes/BinauralWSJ0Mix/separation/train.py:
```diff
@@ -682,13 +709,30 @@ def audio_pipeline_s3(s3_wav):
         )
```
![image](https://user-images.githubusercontent.com/73161104/183464526-d6a4ab6b-a0c8-479d-bac1-be693ed99019.png)
Hi @ycemsubakan, I don't know how to check the scripts. The simliest way is to delete this 'if' command. But I think the dataset generation scripts won't change once this PR merged so we don't need to correct it.

ycemsubakan commented:
@huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine. 

I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.

huangzj421 commented:
> @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> 
> I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.

That's great, thanks! 'Separate PR' means opening a new PR?

ycemsubakan commented:
> > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> 
> That's great, thanks! 'Separate PR' means opening a new PR?

Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.

huangzj421 commented:
> > > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> > 
> > 
> > That's great, thanks! 'Separate PR' means opening a new PR?
> 
> Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.

Okay, got it. But some checks of PR failed, how can we merge it?

ycemsubakan commented:
> > > > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > > > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> > > 
> > > 
> > > That's great, thanks! 'Separate PR' means opening a new PR?
> > 
> > 
> > Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.
> 
> Okay, got it. But some checks of PR failed, how can we merge it?

The checks are failing because of a general problem. It will be fixed soon.

huangzj421 commented:
> > > > > @huangzj421 , all of the .yaml files for me now work. For some reason when I tried with my conda environment, as I mentioned the training was not going down. But with a standard virtual environment it works fine.
> > > > > I think you can do the version check for the dataset in a separate PR, and if everything is fine with @mravanelli, we can merge.
> > > > 
> > > > 
> > > > That's great, thanks! 'Separate PR' means opening a new PR?
> > > 
> > > 
> > > Yes. If you already have a solution for dataset versioning, feel free to push it here. If you think it's going to take time, we can open a new PR, after we merge this one.
> > 
> > 
> > Okay, got it. But some checks of PR failed, how can we merge it?
> 
> The checks are failing because of a general problem. It will be fixed soon.

Okay, thank you!

huangzj421 commented:
Hi @mravanelli , do you have time to check this PR? Cem has reviewed it and everything is fine. Thanks.

ycemsubakan commented:
Everything seems fine to me. I ran the code one other time. I did some updates to README. I will merge once the tests pass.
